### ISSUE TITLE
device mismatch in gather_nd_helper in torch backend

### ISSUE THREAD

**Issue by mohame54** [unifyai/ivy#19612]
### Bug Explanation

gather_nd_helper in torch backend has device mismatch error 

Suggested solution:
def gather_nd_helper(params, indices):
    indices_shape = indices.shape
    params_shape = params.shape
    if len(indices.shape) == 0:
        num_index_dims = 1
    else:
        num_index_dims = indices_shape[-1]
    result_dim_sizes_list = [
        _reduce(mul, params_shape[i + 1 :], 1) for i in range(len(params_shape) - 1)
    ] + [1]
    result_dim_sizes = torch.tensor(result_dim_sizes_list)
    implicit_indices_factor = int(result_dim_sizes[num_index_dims - 1].item())
    flat_params = torch.reshape(params, (-1,))
    new_shape = [1] * (len(indices_shape) - 1) + [num_index_dims]
    indices_scales = torch.reshape(result_dim_sizes[0:num_index_dims], new_shape)
==>indices_scales = indices_scales.to(indices.device)
    indices_for_flat_tiled = torch.reshape(
        torch.sum(indices * indices_scales, -1, keepdim=True), (-1, 1)
    ).repeat(*[1, implicit_indices_factor])
    implicit_indices = torch.unsqueeze(torch.arange(implicit_indices_factor), 0).repeat(
        *[indices_for_flat_tiled.shape[0], 1]
    )
==> implicit_indices = implicit_indices.to(indices.device)
    indices_for_flat = indices_for_flat_tiled + implicit_indices
    flat_indices_for_flat = torch.reshape(indices_for_flat, (-1,)).type(torch.long)
    flat_gather = torch.gather(flat_params, 0, flat_indices_for_flat)
    res = torch.reshape(
        flat_gather, list(indices_shape[:-1]) + list(params_shape[num_index_dims:])
    )
    return res

### Steps to Reproduce Bug

Run the following code:
seq_length = 5
past_key_values_length = 0
pos_ids = ivy.expand_dims(ivy.arange(10), axis=0)
position_ids = pos_ids[:, past_key_values_length : seq_length + past_key_values_length]

### Environment

Google colab

### Ivy Version

v1.1.9

### Backend

- [ ] NumPy
- [ ] TensorFlow
- [X] PyTorch
- [ ] JAX

### Device

NVIDIA T4 GPU

**Comment by DragosStoican** [comment]
Hi @mohame54 , apologies for the late response, this somehow flew under the radar, could you please provide more details on how to reproduce this bug?

**Comment by mohame54** [comment]
seq_length = 5
past_key_values_length = 0
pos_ids = ivy.expand_dims(ivy.arange(10), axis=0)
position_ids = pos_ids[:, past_key_values_length : seq_length + past_key_values_length]

**Comment by mohame54** [comment]
I provided the steps to produce the bug above @DragosStoican
